analyzers/src/SonarAnalyzer.CSharp/Rules/LiteralsShouldNotBePassedAsLocalizedParameters.cs

```diff
@@ -52,84 +51,74 @@ public sealed class LiteralsShouldNotBePassedAsLocalizedParameters : SonarDiagno
+                .Where(x => x.parameter != null && x.syntax != null)
+                .Where(x => IsLocalizable(x.parameter))
+                .Where(x => IsStringLiteral(x.syntax.Expression, context.SemanticModel))
+                .ToList()
+                .ForEach(x => context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, x.syntax.GetLocation())));
         }
 
-        private static bool IsStringLiteral(ExpressionSyntax expression, SemanticModel semanticModel)
+        private static void AnalyzeAssignments(SyntaxNodeAnalysisContext context)
         {
-            if (expression == null)
+            var assignmentSyntax = (AssignmentExpressionSyntax)context.Node;
+            if (context.SemanticModel.GetSymbolInfo(assignmentSyntax.Left).Symbol is IPropertySymbol propertySymbol
+                && IsLocalizable(propertySymbol)
+                && IsStringLiteral(assignmentSyntax.Right, context.SemanticModel)
+                && !CSharpDebugOnlyCodeHelper.IsCallerInConditionalDebug(assignmentSyntax, context.SemanticModel))
             {
-                return false;
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, assignmentSyntax.GetLocation()));
             }
-
-            var constant = semanticModel.GetConstantValue(expression);
-            return constant.HasValue && constant.Value is string;
         }
 
-        private static bool IsLocalizable(ISymbol symbol)
-        {
-            if (symbol?.Name == null)
-            {
-                return false;
-            }
+        private static bool IsStringLiteral(ExpressionSyntax expression, SemanticModel semanticModel) =>
+            expression != null
+            && semanticModel.GetConstantValue(expression) is { HasValue: true } constant
+            && constant.Value is string;
 
-            return symbol.Name.SplitCamelCaseToWords().Any(localizableSymbolNames.Contains) ||
-                symbol.GetAttributes(KnownType.System_ComponentModel_LocalizableAttribute)
-                    .Any(a => a.ConstructorArguments.Any(c => (c.Value as bool?) ?? false));
-        }
+        private static bool IsLocalizable(ISymbol symbol) =>
+            symbol?.Name != null
+            && symbol.GetAttributes(KnownType.System_ComponentModel_LocalizableAttribute) is var localizableAttribute
+            && IsLocalizable(symbol.Name, localizableAttribute);
+
+        private static bool IsLocalizable(string symbolName, IEnumerable<AttributeData> localizableAttribute) =>
+            localizableAttribute.Any(x => AttributeHasConstructorArgument(x, true))
+            || (symbolName.SplitCamelCaseToWords().Any(LocalizableSymbolNames.Contains)
+               && (!localizableAttribute.Any(x => AttributeHasConstructorArgument(x, false))));
+
+        private static bool AttributeHasConstructorArgument(AttributeData attribute, bool expectedValue) =>
```

Comment:
`AttributeHasConstructorArgument`: I think it's clear that it does a check on the attribute due to the parameter type and name. We can rename it to `HasCtorWith`.

We should make it an extension method on `AttibuteData` to enable reuse.